integration-test/Analysis/GradleSpec.hs

```diff
@@ -0,0 +1,67 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+module Analysis.GradleSpec (spec) where
+
+import Analysis.FixtureExpectationUtils (
+  expectProject,
+  withAnalysisOf,
+ )
+import Analysis.FixtureUtils (
+  AnalysisTestFixture (AnalysisTestFixture),
+  FixtureArtifact (FixtureArtifact),
+  FixtureEnvironment (NixEnv),
+ )
+import Path (mkRelDir, reldir)
+import Strategy.Gradle qualified as Gradle
+import Test.Hspec (Spec, aroundAll, describe, it, shouldBe)
+import Types (DiscoveredProjectType (..))
+
+goEnv :: FixtureEnvironment
+goEnv = NixEnv ["gradle"]
+
+springBoot :: AnalysisTestFixture (Gradle.GradleProject)
+springBoot =
+  AnalysisTestFixture
+    "gradle-java"
+    Gradle.discover
+    goEnv
+    Nothing
+    $ FixtureArtifact
+      "https://github.com/spring-projects/spring-boot/archive/refs/tags/v3.0.0-M1.tar.gz"
+      [reldir|gradle/sample/|]
+      $(mkRelDir "spring-boot-3.0.0-M1")
+
+gradleSettingsOnly :: AnalysisTestFixture (Gradle.GradleProject)
+gradleSettingsOnly =
+  AnalysisTestFixture
+    "gradle-java"
+    Gradle.discover
+    goEnv
+    Nothing
+    $ FixtureArtifact
+      "https://docs.gradle.org/current/samples/zips/sample_building_java_applications-groovy-dsl.zip"
```

Comment:
[nit]: Use a link that's less likely to rot.
```suggestion
      "https://docs.gradle.org/7.3.3/samples/zips/sample_building_java_applications-groovy-dsl.zip"
```

Reply:
Good idea, thanks!